### PR TITLE
mpress: add version test, clean up

### PR DESCRIPTION
**Pull request by L2G**
* Add test to verify version of executable
* Other changes for `brew audit --strict`

### CODE REVIEW

**Comment by L2G** [comment]
From Jenkins:

```
==> brew test --verbose Homebrew/binary/mpress				 FAILED
Testing homebrew/binary/mpress
==> /usr/local/Cellar/mpress/1.21/bin/mpress
Error: homebrew/binary/mpress: failed
<1> expected but was
<nil>.
```

Anyone have any idea why that would happen on 10.8 and 10.10 but not 10.9?

**Comment by mistydemeo** [comment]
Running locally:

```
$ mpress --version
fish: 'mpress --version' terminated by signal SIGKILL (Forced quit)
```

**Comment by mistydemeo** [comment]
I can confirm in some VMs from 10.8 through 10.10: it runs only on 10.9, and crashes instantly on other OS versions.

**Comment by mistydemeo** [comment]
```
lldb -c /cores/core.8905
(lldb) target create --core "/cores/core.8905"
warning: (x86_64) /cores/core.8905 load command 38 LC_SEGMENT_64 has a fileoff + filesize (0x1af69000) that extends beyond the end of the file (0x1af68000), the segment will be truncated to match
warning: (x86_64) /cores/core.8905 load command 39 LC_SEGMENT_64 has a fileoff (0x1af69000) that extends beyond the end of the file (0x1af68000), ignoring this section
Core file '/cores/core.8905' (x86_64) was loaded.
Process 0 stopped
* thread #1: tid = 0x0000, 0x00007fff5fc01000 dyld`_dyld_start, stop reason = signal SIGSTOP
    frame #0: 0x00007fff5fc01000 dyld`_dyld_start
```

This may be an issue only with the x86_64 slice; after splitting it into separate i386 and x86_64 binaries, only the x86_64 binary crashes.

**Comment by mistydemeo** [comment]
Maybe change the install method like so so we install a working 32-bit only binary:

```patch
--- a/mpress.rb
+++ b/mpress.rb
@@ -9,6 +9,7 @@ class Mpress < Formula
   depends_on :arch => :intel
 
   def install
-    bin.install "mmpress" => "mpress"
+    bin.mkpath
+    system "lipo", "-thin", "i386", "mmpress", "-o", "#{bin}/mpress"
   end
 end
```

**Comment by mistydemeo** [comment]
Actually, it looks like the issue is that `mpress`-compressed x86_64 executables are failing like this. I can reproduce this behaviour on any x86_64 binary (thin or fat) that I compress using `mpress`.

**Comment by mistydemeo** [comment]
This seems fairly severe, and the lack of reports since this was added over a year ago implies to me it's widely used. Thinking either:

a) We need to add a note to the caveats that `mpress` only produces working 32-bit executables, or
b) The formula could get moved to the boneyard.

I don't mind the former since it *does* work, albeit with significant limitations.

**Comment by L2G** [comment]
@mistydemeo: I'm going to incorporate your changes. Since a distinctly different binary will be installed, I'm assuming a revision bump is in order, too.

I don't really have a horse in this race. I just wanted to add a test to check the version number, in the unlikely event that a new release gets published; then the test should catch if someone updates the SHA hash without updating the version.

Thanks for the huge help with this!

**Comment by mistydemeo** [comment]
Sounds good! Let's keep this for now then, with the changes.

File: mpress.rb [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
+  # https://github.com/Homebrew/homebrew-binary/pull/252
   def install
-    bin.install "mmpress" => "mpress"
+    bin.mkpath
+    system "lipo", "-thin", "i386", "mmpress", "-o", "#{bin}/mpress"
+  end
+
+  def caveats; <<-EOS.undent
+    Because of intractable x86_64 issues, the installed mpress binary is
+    stripped down to i386 code, and thus it can only compress other i386
```
Comment (mistydemeo) [inline mpress.rb]: Can remove the "thus"; the i386 binary *can* compress x86_64 binaries, they're just broken (like the x86_64 slice of mpress itself).

**Comment by mistydemeo** [comment]
Thanks, merged!

**Pull request closed without merging**